### PR TITLE
Jpeg mode

### PR DESCRIPTION
**Pull request by jepler**
.. including an example for Kaluga which captures to the internal CIRCUITPY flash region.

### CODE REVIEW

**Comment by jepler** [comment]
@ladyada do you want to test this yourself? do you want an example that uses SD card instead of CIRCUITPY? (shouldn't be too difficult, just would need to wire it up)

**Comment by ladyada** [comment]
i dont need to test it - thankx! please check the OV2640 datasheet, anything else interesting in there? filters?

File: adafruit_ov2640.py [hunk 1]
```diff
@@ -886,21 +887,36 @@
...
-#     0xD9, 0x10,
-#     0xDF, 0x80,
-#     0x33, 0x80,
-#     0x3C, 0x10,
-#     0xEB, 0x30,
-#     0xDD, 0x7F,
-#     _RESET, 0x00,
-# ])
+_ov2640_settings_jpeg3 = bytes(
+    [
```
Comment (tannewt) [inline adafruit_ov2640.py:891]: If you want to save import memory then you could switch this to an ugly bytes literal. As-is it's a list construction that is then converted to bytes.

File: adafruit_ov2640.py [hunk 2]
```diff
@@ -1140,17 +1169,22 @@ def colorspace(self):
...
-        self._write_list(
-            _ov2640_settings_rgb565
-            if colorspace == OV2640_COLOR_RGB
-            else _ov2640_settings_yuv422
-        )
+        self._set_size_and_colorspace()
+
+    def _set_colorspace(self):
+        colorspace = self._colorspace
+        if colorspace == OV2640_COLOR_RGB:
```
Comment (tannewt) [inline adafruit_ov2640.py:1176]: Want to use a dictionary to lookup the corresponding settings?

**Review by tannewt — CHANGES_REQUESTED** [review]
Two minor things you might want to do. I'm ok if you don't want to do them.

File: adafruit_ov2640.py (hunk 2 above)
Comment (jepler) [reply adafruit_ov2640.py]: done!

**Comment by jepler** [comment]
@tannewt ready for a re-review.  I added a 2nd jpeg demo which turns the Kaluga into a selfie-cam.  Press (hold, really, because it's polled) the REC button to save a jpeg image at highest resolution to an SD card.

img0003

**Review by tannewt — APPROVED** [review]
Looks great! Thank you!

**Comment by ladyada** [comment]
nice work!

**Pull request merged**